### PR TITLE
Benchmark

### PR DESCRIPTION
**Pull request by SokolovVadim**

### CODE REVIEW

File: radix/insert_test.go [hunk 1]
```diff
@@ -38,19 +34,21 @@ func CreateSubstrings(str string) []string {
...
 		// fmt.Println(substringArray[i], " inserted")
 	}
 }
 
-// go test -bench=. -benchmem
+// go test -bench=. -benchmem -benchtime=100x
 func BenchmarkInsert(b *testing.B) {
 	InitSeed()
-	test_str := GenerateTestString(Lenght)
+	test_str := GenerateTestString(b.N)
```
Comment (neganovalexey) [inline radix/insert_test.go]: остроумно, но лучше b.N раз создать индекс с нуля

File: radix/get_test.go [hunk 2]
```diff
@@ -1,24 +1,33 @@
 package radix_test
 
 import (
-	"testing"
 	"github.com/SokolovVadim/Radix-Tree"
+	"testing"
+)
+
+const (
+	Length = 65536;
```
Comment (neganovalexey) [inline radix/get_test.go]: why do you need to export the constants?

File: radix/get_test.go [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
+	RightPos = 1032;
 )
 
-func GetSubstring(b *testing.B, r *radix.Tree, substringArray []string) {
-	for i := 0; i < len(substringArray); i++ {
-		_, ok := r.Get(substringArray[i])
-		if !ok {
-			b.Fatalf("missing key: %v", substringArray[i])
-		}
+func GetSubstring(b *testing.B, r *radix.Tree, testStr string) {
```
Comment (neganovalexey) [inline radix/get_test.go]: same: private functions should start their name with lowercase character

File: radix/get_test.go [hunk 4]
```diff
@@ -1,33 +1,46 @@
...
+	var pos interface{}
+	// there is only one possible match for a string
+	fn := func(s string, v interface{}) bool {
+		out, pos = s, v
+		return false
 	}
+
+	r.WalkPrefix(subString, fn)
+	// fmt.Println("out:", out, "pos =", pos)
+	if !reflect.DeepEqual(out, testStr[LeftPos: Length]) {
```
Comment (neganovalexey) [inline radix/get_test.go]: better to cast `out` to string, i. e.
```
if result, ok := out.(string); !ok || result != testStr[left:length] {
```

File: suffix_array/suffix_test.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func GetSubstr(sa* suffixarray.Index, joinedStrings string,
+	           subString string) (string, int, int) {
+	// fmt.Println("substr:", subString)
+	match, err := regexp.Compile("\x00" + subString + "[^\x00]*")
+	if err != nil {
+		panic(err)
+	}
+	ms := sa.FindAllIndex(match, -1)
```
Comment (neganovalexey) [inline suffix_array/suffix_test.go]: why Lookup is insufficient?

File: radix/insert_test.go (hunk 1 above)
Comment (SokolovVadim) [reply radix/insert_test.go]: Testing flag `-benchtime=100x` allows to perform 100 tests, but it does not mean that `b.N` is always equal to 100. Referring to a [documentation](https://golang.org/pkg/testing/):  "_During benchmark execution, b.N is adjusted until the benchmark function lasts long enough to be timed reliably_". Tests show that at the very beginning of execution b.N is equal to 1.

File: radix/get_test.go (hunk 2 above)
Comment (SokolovVadim) [reply radix/get_test.go]: fixed

File: suffix_array/suffix_test.go (hunk 5 above)
Comment (SokolovVadim) [reply suffix_array/suffix_test.go]: changed `FindAllIndex()` to `Lookup()`

**Review by neganovalexey — APPROVED** [review]

**Pull request merged**